service/src/main/java/org/databiosphere/workspacedataservice/shared/model/job/Job.java

```diff
@@ -68,6 +68,19 @@ public Job(
     this.result = result;
   }
 
+  /**
+   * Convenience for creating a new Job object, using a random UUID as jobId and defaults for job
+   * status
+   *
+   * @param jobType type of the job
+   * @param input input arguments for the job
+   * @return the job, after creation
+   */
+  public static Job<JobInput, JobResult> newJob(JobType jobType, JobInput input) {
+    LocalDateTime now = LocalDateTime.now();
+    return new Job<>(UUID.randomUUID(), jobType, JobStatus.CREATED, null, now, now, input, null);
```

Comment:
Small readability thing I like to do with large param lists, in the absence of named parameters: add an explainer comment for any arg that doesn't have a self-documenting name.  (This suggestion might need some re-wrapping to look nice)

```suggestion
    return new Job<>(UUID.randomUUID(), jobType, JobStatus.CREATED, /* errorMessage=*/ null, /* created=*/ now, /* updated=*/ now, input, /* result=*/ null);
```